### PR TITLE
Cloud corfu: bring logback config, and JAVA_OPTS into docker image

### PR DESCRIPTION
**Pull request by xnull**
## Overview

Description:
Cloud Corfu: bring logback config, and JAVA_OPTS into the docker image

Why should this be merged: 

Related issue(s) (if applicable): #<number>

## Checklist (Definition of Done):

- [x] There are no TODOs left in the code
- [x] [Coding conventions](https://github.com/CorfuDB/CorfuDB/wiki/Corfu-Style-Guidelines) (e.g. for logging, unit tests) have been followed
- [x] Change is covered by automated tests
- [x] Public API has Javadoc

### CODE REVIEW

File: infrastructure/Dockerfile [hunk 1]
```diff
@@ -15,8 +15,29 @@ COPY target/${CORFU_TOOLS_JAR} /usr/share/corfu/lib/${CORFU_TOOLS_JAR}
 COPY target/bin /usr/share/corfu/bin
 COPY target/corfu_scripts /usr/share/corfu/corfu_scripts
 COPY target/scripts /usr/share/corfu/scripts
+COPY target/logback.prod.xml usr/share/corfu/conf
```
Comment (SravanthiAshokKumar) [inline infrastructure/Dockerfile:18]: is a '/' missing before 'usr/'?

Comment (xnull) [reply infrastructure/Dockerfile]: yep, you are right
fixed

File: infrastructure/Dockerfile [hunk 2]
```diff
@@ -15,8 +15,31 @@ COPY target/${CORFU_TOOLS_JAR} /usr/share/corfu/lib/${CORFU_TOOLS_JAR}
...
+    -XX:NumberOfGCLogFiles=10 \
+    -XX:GCLogFileSize=5M \
+    -XX:+HeapDumpOnOutOfMemoryError \
+    -XX:HeapDumpPath=/var/log/corfu/corfu_oom.hprof \
+    -XX:+HeapDumpOnOutOfMemoryError \
+    -Djdk.nio.maxCachedBufferSize=1048576 \
+    -Dio.netty.recycler.maxCapacityPerThread=0 \
+    -XX:+PrintGCApplicationStoppedTime \
+    -XX:+PrintGCApplicationConcurrentTime \
+    -Djava.io.tmpdir=/image/corfu-server/temp
```
Comment (hisundar) [inline infrastructure/Dockerfile:40]: `-d DEBUG -k 5000000 --log-size-quota-percentage=40 --metrics -l /config -q eth0 -c 0.3`
Basically can we ensure all the current parameters that corfu server is started with are retained as much as possible?

Comment (xnull) [reply infrastructure/Dockerfile:40]: In the docker container, I can only define JAVA_OPTS, everything is defined in helm charts: https://github.com/CorfuDB/corfudb-cloud/blob/c6ab4066223e6bee7a0ffd8852ffbb7fac21342e/cloud/corfu/corfu/templates/statefulset.yaml#L37

Comment (hisundar) [reply infrastructure/Dockerfile:40]: but what happens if the corfu docker container is started without helm ?

**Review by hisundar — APPROVED** [review]

**Pull request merged**